src/fileio/fileio_scen.cpp

```diff
@@ -155,6 +155,33 @@ template<typename Container> static void port_shop_spec_node(cSpecial& spec, std
 	spec.ex2a = spec.ex2b = -1;
 }
 
+static void port_special_text_response(cSpecial &special, cScenario& scenario)
```

Comment:
Can you explain what this is doing and why?

Reply:
In a legacy scenario, all the strings are stored in one block. But load_scenario_v1 split them in different variables
`
	for(short i = 0; i < 270; i++) {
		len = (long) (temp_scenario.scen_str_len[i]);
		fread(temp_str, len, 1, file_id);
		temp_str[len] = 0;
		if(i == 0) scenario.scen_name = temp_str;
		else if(i == 1 || i == 2)
			scenario.who_wrote[i-1] = temp_str;
		else if(i == 3)
			scenario.contact_info[1] = temp_str;
                ...
`
So, after uploading the scenario, we need to check that a string referenced by a special node was not referencing an unusual string (and if so, correct the problem).

Reply:
Is it actually possible for a special node to reference an unusual string? Have you confirmed that this works in  legacy BoE?